### PR TITLE
fix(desktop/browser): Fix add bookmark button

### LINKED ISSUE #4108: Add Favorite button does nothing
# Bug Report

## Description

Can't add to Favorite

## Steps to reproduce

1. open browser
2. enable Fav bar
3. open any URL for example google.com
4. click Add Favorite

#### Expected behavior

Add Favorite modal opens

#### Actual behavior

https://user-images.githubusercontent.com/82375995/141442678-930513a0-df71-4569-a1f8-553a127282d9.mov

### Additional Information

- Status desktop version: master
- Operating System: masOS Big Sur

### PR DESCRIPTION
**Pull request by alaibe**
fixes #4108

### What does the PR do
This button needs special case when it is not a real bookmark

### Affected areas

browser

### Screenshot of functionality

image

### CODE REVIEW

**Review by Khushboo-dev-cpp — APPROVED** [review]

File: ui/app/AppLayouts/Browser/panels/FavoritesBar.qml [hunk 1]
```diff
@@ -35,18 +35,28 @@ RowLayout {
                 acceptedButtons: Qt.LeftButton | Qt.RightButton
                 z: 51
                 onClicked: function (mouse) {
+                    const isAddBookmarkButton = url === " "
```
Comment (jrainville) [inline ui/app/AppLayouts/Browser/panels/FavoritesBar.qml:38]: I don't remember how that works. Where does `url` come from and how come it is a space when it's the bookmark button?

Comment (alaibe) [reply ui/app/AppLayouts/Browser/panels/FavoritesBar.qml:38]: Yeah, it is quite weird.... It took me a bit of time to understand. Here is how I think it works: The add bookmark button is a bookmark saved with the url set at empty. So when we display the bookmark in the toolbar we have a special case for the empty space which correspond to the button to add a bookmark

Comment (Khushboo-dev-cpp) [reply ui/app/AppLayouts/Browser/panels/FavoritesBar.qml:38]: Yea I did that part of adding a new item to the model with " " string. This in turn represents the "add fav button". I did this to handle the requirement that the addFav button should always be at the end after a bookmark is added/deleted.

Comment (jrainville) [reply ui/app/AppLayouts/Browser/panels/FavoritesBar.qml:38]: Ah ok, makes sense. What I would recommend then is to replace the space with a constant. Something like `NEW_BOOKMARK = "ADD_NEW_BOOKMARK"`
That way, it's clearer for us, and also is easier to expend if we need to send it from multiple places.

Comment (alaibe) [reply ui/app/AppLayouts/Browser/panels/FavoritesBar.qml:38]: np, I will do the constant, but that will be a breaking change to rename it to Add new bookmark right? cause people will start to see an empty bookmark. @jrainville ok with that?

Comment (jrainville) [reply ui/app/AppLayouts/Browser/panels/FavoritesBar.qml:38]: Oh crap, I misunderstood the behavior.
Ok then the constant is still a good idea, but make the constant equal the space. So `NEW_BOOKMARK = " "`

Comment (alaibe) [reply ui/app/AppLayouts/Browser/panels/FavoritesBar.qml]: Updated!

Comment (jrainville) [reply ui/app/AppLayouts/Browser/panels/FavoritesBar.qml]: I think you just need to also update the `addBookmark(" ")` and `deleteBookmark(" ")` calls throughout the code to use the Contsant

Comment (alaibe) [reply ui/app/AppLayouts/Browser/panels/FavoritesBar.qml]: you're right, all updated this time (I think)

**Review by jrainville — APPROVED** [review]
All good. Nice job

**Pull request merged**